Modules/_io/bufferedio.c

```diff
@@ -842,7 +847,13 @@ _io__Buffered_peek_impl(buffered *self, Py_ssize_t size)
     PyObject *res = NULL;
 
     CHECK_INITIALIZED(self)
-    CHECK_CLOSED(self, "peek of closed file")
+
+    // use a check specific to peek if the previous file operation was not a read
+    if (Py_SAFE_DOWNCAST(self->pos, Py_off_t, Py_ssize_t) == 0) {
```

Comment:
If I'm reading this correctly, this just means we haven't done a read or a seek yet?

Wouldn't it be better to not close the file after a peek, but wait for the read? If the read never comes, it'll be closed when the stream is closed as it normally would.

Reply:
> If I'm reading this correctly, this just means we haven't done a read or a seek yet?
thanks for your comment, and yes the if condition checks if a read or a seek has been done before.

> Wouldn't it be better to not close the file after a peek, but wait for the read? If the read never comes, it'll be closed when the stream is closed as it normally would.
I considered this solution initially as well, but unfortunately based on the [example provided](https://bugs.python.org/issue44687), the closing of the file is actually done by one of the methods of the object that `requests.get("https://amazon.com", stream=True).raw` returns. 

I've just pushed an alternate fix, where I've reset some attributes of BufferedReader when its method `closed()` has been called. I believe this fixes the bug while passing the test case I've previously encountered which prompted me to write this i condition in the first place. Do let me know what you think.